sn_networking/src/event/kad.rs

```diff
@@ -395,9 +399,40 @@ impl SwarmDriver {
                     Self::send_record_after_checking_target(sender, peer_record.record, &cfg)?;
                 } else {
                     debug!("For record {pretty_key:?} task {query_id:?}, fetch completed with split record");
-                    sender
-                        .send(Err(GetRecordError::SplitRecord { result_map }))
-                        .map_err(|_| NetworkError::InternalMsgChannelDropped)?;
+                    let mut accumulated_spends = BTreeSet::new();
+                    for (record, _) in result_map.values() {
+                        match get_raw_signed_spends_from_record(record) {
+                            Ok(spends) => {
+                                accumulated_spends.extend(spends);
+                            }
+                            Err(_) => {
+                                continue;
+                            }
+                        }
+                    }
+                    if !accumulated_spends.is_empty() {
+                        info!("For record {pretty_key:?} task {query_id:?}, found split record for a spend, accumulated and sending them as a single record");
+                        let accumulated_spends = accumulated_spends
+                            .into_iter()
+                            .take(2)
+                            .collect::<Vec<SignedSpend>>();
+
+                        let bytes = try_serialize_record(&accumulated_spends, RecordKind::Spend)?;
+
+                        let new_accumualted_record = Record {
+                            key: peer_record.record.key,
+                            value: bytes.to_vec(),
+                            publisher: None,
+                            expires: None,
+                        };
+                        sender
+                            .send(Ok(new_accumualted_record))
+                            .map_err(|_| NetworkError::InternalMsgChannelDropped)?;
+                    } else {
+                        sender
+                            .send(Err(GetRecordError::SplitRecord { result_map }))
```

Comment:
if no entry within the accumulated_record,
it shall be considered as NoHolders, instead of `SplitRecord` ?

Reply:
So we have no idea of the record we're GET'ing is a spend here, we just try to read the header and collect them IF they're spends. If not, we assume it is a chunk/register and just return split record.